src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -399,6 +354,48 @@ bool ProcessPrimaryQueue ()
 			return true;
 		}
 
+		bool ProcessMarkedPending ()
+		{
+			bool marked = false;
+			foreach (var pending in Annotations.MarkedPending ()) {
+				marked = true;
+
+				// Some pending items might be processed by the time we get to them.
+				if (Annotations.IsProcessed (pending))
+					continue;
+
+				switch (pending) {
+				case TypeDefinition type:
+					MarkType (type, DependencyInfo.AlreadyMarked, type);
+					break;
+				case MethodDefinition method:
+					MarkMethod (method, DependencyInfo.AlreadyMarked, null);
+					// Methods will not actually be processed until we drain the method queue.
+					break;
+				case FieldDefinition field:
```

Comment:
Currently they're not handled by Initialize, so I kept it that way but made it an exception instead. A step that marks a property/event would also need to mark the methods like we do in ResolveFromXml, but marking the methods will already mark their properties/events, so it would be a little redundant. There are some other options:
- Allow marking properties/events, and mark all of their methods when processing pending items
  - Custom steps which only want to mark some property methods (like we do for XML) would want to avoid it.
- Allow marking properties/events, but don't do any additional processing for them
  - I am not sure if it would ever make sense to mark properties/events without any of their methods.

Reply:
Well - technically for example pure serializer would need to only mark getters, not setters, but it's probably unnecessary optimization.

I think that making it throw is a good idea - it's basically unsupported (as it won't do anything).

Reply:
With the current behavior they could just mark the getter method, and MarkStep would take care of the property. They just need to avoid marking the Property directly.